sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -565,11 +565,14 @@ case class StringToMap(text: Expression, pairDelim: Expression, keyValueDelim: E
   extends TernaryExpression with ExpectsInputTypes {
   override def nullIntolerant: Boolean = true
   def this(child: Expression, pairDelim: Expression) = {
-    this(child, pairDelim, Literal(":"))
+    this(child, pairDelim, Literal.create(":", SQLConf.get.defaultStringType))
```

Comment:
I don't quite get this change. Should this be default collation?

Reply:
I moved this to the `Literal` itself, but yes the literals should produce default collation. However, the final state of this depends on the other PR for resolving default string types.

Reply:
Are you saying that the string literal here must use the session collation? I think the previous code makes more sense as it uses `object StringType`, which we can determine the default collation later on.

Reply:
I completely agree, however right now we don't have any logic in master branch that determines default collation and we are just using session collation. But I am addressing this in the other PR.

Reply:
so this is a temporary change and we will revert it when the other PR gets merged?

Reply:
Yes